### ISSUE TITLE
[p0] Set attention mask and pad token in beam generate

### ISSUE THREAD

**Issue by annh3** [annh3/ai_planning_searching#4]
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.

Replicate the issue by running mcts_utils_tests.py

**Comment by annh3** [comment]
https://discuss.huggingface.co/t/finetuning-whisper-attention-mask-not-set-and-canot-be-inferred/97456/2

**Comment by annh3** [comment]
https://github.com/sytelus/jupyter_nbs/blob/main/codegen_decoding.ipynb

**Comment by annh3** [comment]
Can replicate in the python console like 

```
>>> pretrained_weights = 'gpt2'
>>> tokenizer = GPT2TokenizerFast.from_pretrained(pretrained_weights)
/Users/annhe/anaconda3/envs/agency/lib/python3.10/site-packages/transformers/tokenization_utils_base.py:1601: FutureWarning: `clean_up_tokenization_spaces` was not set. It will be set to `True` by default. This behavior will be depracted in transformers v4.45, and will be then set to `False` by default. For more details check this issue: https://github.com/huggingface/transformers/issues/31884
  warnings.warn(
>>> (tokenizer.eos_token, tokenizer.pad_token)
('<|endoftext|>', None)
```

**Comment by annh3** [comment]
https://discuss.huggingface.co/t/do-automatically-generated-attention-masks-ignore-padding/15479

**Comment by annh3** [comment]
According to [this](https://github.com/suno-ai/bark/issues/402) the warning should not cause too many issues, so closing this. Will re-open if there are observed issues with text generation :).
